app/models/qernel/graph.rb

```diff
@@ -1,450 +1,438 @@
+    #      converter_1.ready?     # => true
+    #      converter_1.calculate
+    #
+    #
+    def optimize_calculation_order
+      copy = DeepClone.clone self #Marshal.load(Marshal.dump(self))
+      copy.calculate
+      copy.finished_converters.reverse.each_with_index do |converter, index|
+        if old = converters.detect{|c| c.id == converter.id}
+          converters.delete(old)
+          converters.unshift(old)
+        end
       end
     end
-  end
 
-  # --- Goal-related methods --------------------------------------------------
+    # --- Goal-related methods --------------------------------------------------
 
 
-  # Returns an array with all the defined goals. The value is not memoized because
-  # goals can be added dynamically
-  #
-  def goals
-    @goals ||= []
-  end
+    # Returns an array with all the defined goals. The value is not memoized because
+    # goals can be added dynamically
+    #
+    def goals
+      @goals ||= []
+    end
 
-  # Returns a goal by key
-  #
-  def goal(key)
-    goals.find {|g| g.key == key}
-  end
+    # Returns a goal by key
+    #
+    def goal(key)
+      goals.find {|g| g.key == key}
+    end
 
-  # finds or create goal as needed
-  #
-  def find_or_create_goal(key)
-    unless g = goal(key)
-      g = Goal.new(key)
-      goals << g
+    # finds or create goal as needed
+    #
+    def find_or_create_goal(key)
+      unless g = goal(key)
+        g = Goal.new(key)
+        goals << g
+      end
+      return g
     end
-    return g
-  end
 
-  def reset_goals
-    @goals = []
-  end
+    def reset_goals
+      @goals = []
+    end
 
-  def reset_memoized_methods
-    @carriers = nil
-    @links    = nil
-    @groups   = nil
-    @converters_by_group = {}
+    def reset_memoized_methods
+      @carriers = nil
+      @links    = nil
+      @groups   = nil
+      @converters_by_group = {}
 
-    regenerate_converter_lookup
-  end
+      regenerate_converter_lookup
+    end
 
-  # Regenerate a lookup hash for converters, used by #converter( key_or_excel_id )
-  #
-  def regenerate_converter_lookup
-    @converters_hash = {}
+    # Regenerate a lookup hash for converters, used by #converter( key_or_excel_id )
+    #
+    def regenerate_converter_lookup
+      @converters_hash = {}
 
-    converters.each do |converter|
-      @converters_hash[converter.key] = converter
+      converters.each do |converter|
+        @converters_hash[converter.key] = converter
+      end
     end
-  end
-
-  def graph_query
-    @graph_query ||= GraphApi.new(self)
-  end
 
-  def initializer_inputs
-    decorated_inputs.
-      sort_by { |input, _| input.priority }.
-      reverse
-  end
+    def graph_query
+      @graph_query ||= GraphApi.new(self)
+    end
 
-  def decorated_inputs
-    (area.init || {}).map do |input_key, input_value|
-      [Input.fetch(input_key), input_value]
+    def initializer_inputs
+      @initializer_inputs ||= InitializerInputs.for_area(area.area_code)
```

Comment:
I am not sure whether this will - unintentionally - survive a cache flush, e.g. an Etsource::Reloader call.